pandaserver/dataservice/setupper_atlas_plugin.py

```diff
@@ -0,0 +1,2226 @@
+                                # backend
+                                ddm_back_end = job.getDdmBackEnd()
+                                if ddm_back_end is None:
+                                    ddm_back_end = "rucio"
+                                # register dataset
+                                self.logger.debug(f"registerNewDataset {name} metadata={tmp_metadata}")
+                                is_ok = False
+                                for _ in range(3):
+                                    try:
+                                        out = rucioAPI.register_dataset(
+                                            name,
+                                            metadata=tmp_metadata,
+                                            lifetime=tmp_life_time,
+                                        )
+                                        self.logger.debug(out)
+                                        new_vuid = out["vuid"]
+                                        is_ok = True
+                                        break
+                                    except Exception:
+                                        error_type, error_value = sys.exc_info()[:2]
+                                        self.logger.error(f"registerDataset : failed with {error_type}:{error_value}")
+                                        time.sleep(10)
+                                if not is_ok:
+                                    tmp_msg = f"Setupper._setupDestination() could not register : {name}"
+                                    dest_error[dest] = tmp_msg
+                                    self.logger.error(tmp_msg)
+                                    continue
+                                # register dataset locations
+                                if (
+                                    job.lockedby == "jedi" and job.getDdmBackEnd() == "rucio" and job.prodSourceLabel in ["panda", "user"]
+                                ) or DataServiceUtils.getDistributedDestination(file.destinationDBlockToken, ignore_empty=False) is not None:
+                                    # skip registerDatasetLocations
+                                    status, out = True, ""
+                                elif (
+                                    name == original_name
+                                    or tmp_src_ddm != tmp_dst_ddm
+                                    or job.prodSourceLabel == "panda"
+                                    or (
+                                        job.prodSourceLabel in JobUtils.list_ptest_prod_sources
+                                        and job.processingType in ["pathena", "prun", "gangarobot-rctest"]
+                                    )
+                                    or len(tmp_token_list) > 1
+                                    or EventServiceUtils.isMergeAtOS(job.specialHandling)
+                                ):
+                                    # set replica lifetime to _sub
```

Comment:
replica lifetimes 7 vs 14 days could be documented a bit and be extracted as constants somewhere